pages/_app.js

```diff
@@ -35,7 +35,13 @@ export default function App({ Component, pageProps }) {
     );
     router.push(`/PetDetails/${updatedPetData.id}`);
   }
-
+  function handleInteractPet(updatedPetData) {
+    setPetCollection((prevData) =>
+      prevData.map((pet) =>
+        pet.id === updatedPetData.id ? { ...pet, ...updatedPetData } : pet
+      )
+    );
+  }
```

Comment:
Handling every pet interactions like this can potentially cause performance issues, especially if your `petCollection` is large or the interaction happens frequently. 

The way you're currently handling the update `setPetCollection` re-renders the entire list every time a single pet is updated -  with every interaction. 


Consider using a more targeted approach, such as updating only the specific pet in the state directly. There are multiple ways to achieve this. 

While the current method works for now, please keep this in mind as a potential optimisation if performance issues arise in the future.

Reply:
Thank you, as per checkout today we will adress this within future Stories